user/build-stages/share-files-s3.md

```diff
@@ -11,14 +11,17 @@ This example has 2 build stages:
 Here's what the `.travis.yml` config could look like:
 
 ```yaml
+dist: xenial
+
 env:
   global:
     # include $HOME/.local/bin for `aws`
     - PATH=$HOME/.local/bin:$PATH
 
 before_install:
-  # set up awscli packages
-  - pip install --user awscli
+  - pyenv global 3.7.1
```

Comment:
Why not just put at the top of the script:
```
language: python
python: 3.7
```

Reply:
I think the example is meant to work regardless of the language setting, because people may want to use the technique on projects in various languages.

Reply:
Thanks.  Now I understand.  Pyenv is superslow so what about this approach on Xenial or Bionic?
```
- python3 -m pip || wget -qO- https://bootstrap.pypa.io/get-pip.py | sudo -H python3
- python3 -m pip install --user awscli
- aws --version  # aws-cli/1.16.217 Python/3.5.2 Linux/4.15.0-1028-gcp botocore/1.12.207
```

Reply:
> No module named pip

I set up a tiny [test repository](https://github.com/takluyver/travis-test/) to experiment with what is possible on Travis. I couldn't find any way to use pip on Python 3 without either invoking pyenv or installing pip first - I assume either of those options is unwanted hassle. Feel free to play around and see if there's something I missed, though.

So I think before this example can be changed, it would be necessary to change the default Travis build images, e.g. to include the `python3-pip` package.

Reply:
* https://travis-ci.com/cclauss/awscli-on-xenial-hack
* https://github.com/cclauss/awscli-on-xenial-hack/blob/master/.travis.yml

Reply:
Yup, it works if you install pip first. But I'd imagine people won't especially want an extra install-the-installer step given that the example as it stands works without that.

Reply:
pyenv takes a minute and a half to build Python 3.

Reply:
I mean, compared to the example as it currently is, using Python 2.

Reply:
I had a go at adding pip-on-Python 3 to the default images, but I don't really know the infrastructure, I just found a likely looking bit of code:

https://github.com/travis-ci/travis-cookbooks/pull/1055